packages/frontend/src/components/navigation/Navigation.js

```diff
@@ -36,114 +29,85 @@ const Container = styled.div`
         const mobileMenu = document.getElementById('mobile-menu');
 
         if (e.target.tagName === 'SPAN') {
             return false;
         }
 
-        if (e.target.tagName === 'BUTTON' || e.target.tagName === 'A' || (!desktopMenu.contains(e.target) && !mobileMenu.contains(e.target))) {
-            this.setState({ menuOpen: false });
+        if (e.target.tagName === 'BUTTON' || e.target.tagName === 'A' || (!desktopMenu?.contains(e.target) && !mobileMenu?.contains(e.target))) {
+            setMenuOpen(false);
         }
-    }
+    }, []);
 
-    get showNavLinks() {
-        return this.props.account.localStorage?.accountFound;
-    }
-
-    toggleMenu = () => {
-        if (!this.state.menuOpen) {
-            this.props.getAvailableAccountsBalance();
+    const toggleMenu = useCallback(() => {
+        if (menuOpen) {
+            setMenuOpen(false);
+        } else {
+            setMenuOpen(true);
         }
-
-        this.setState((prevState) => ({
-            menuOpen: !prevState.menuOpen
-        }));
-    }
-
-    handleSelectAccount = (accountId) => {
-        this.props.switchAccount({ accountId });
-        this.setState({ menuOpen: false });
-    }
-
-    render() {
-        const { menuOpen } = this.state;
-        const { 
-            flowLimitationMainMenu,
-            flowLimitationSubMenu
-        } = this.props;
-
-        return (
-            <Container id='nav-container' open={menuOpen}>
-                <DesktopContainer
-                    menuOpen={menuOpen}
-                    toggleMenu={this.toggleMenu}
-                    handleSelectAccount={this.handleSelectAccount}
-                    showNavLinks={this.showNavLinks}
-                    flowLimitationMainMenu={flowLimitationMainMenu}
-                    flowLimitationSubMenu={flowLimitationSubMenu}
-                    refreshBalance={this.props.getAccountBalance}
-                    {...this.props}
-                />
-                <MobileContainer
-                    menuOpen={menuOpen}
-                    toggleMenu={this.toggleMenu}
-                    handleSelectAccount={this.handleSelectAccount}
-                    showNavLinks={this.showNavLinks}
-                    flowLimitationMainMenu={flowLimitationMainMenu}
-                    flowLimitationSubMenu={flowLimitationSubMenu}
-                    refreshBalance={this.props.getAccountBalance}
-                    {...this.props}
-                />
-            </Container>
-        );
-    }
-}
-
-const mapStateToProps = (state) => ({
-    account: selectAccountSlice(state),
-    router: getRouter(state),
-    availableAccounts: selectAvailableAccounts(state),
-    flowLimitationMainMenu: selectFlowLimitationMainMenu(state),
-    flowLimitationSubMenu: selectFlowLimitationSubMenu(state)
-});
-
-const mapDispatchToProps = {
-    switchAccount,
-    getAvailableAccountsBalance,
-    getAccountBalance
+    }, [menuOpen]);
+
+    const handleSelectAccount = (accountId) => {
```

Comment:
:nit: this should also be wrapped in `useCallback()`

Reply:
Added :)